plugins/knife/knife-deployment-resize.rb

```diff
@@ -287,6 +289,36 @@ def get_current_counts
       return result
     end
 
+    def update_ralf_hostname environment, cloud
+      ralfs = find_nodes(roles: "clearwater-infrastructure", role: "ralf").length
+
+      changed_nodes = []
+
+      %w{bono ibcf sprout ralf}.each do |node_type|
+        find_nodes(roles: "clearwater-infrastructure", role: node_type).each do |node|
+          has_ralf = node.set[:clearwater][:ralf]
+          puts "#{node.name}: ralf attribute is #{has_ralf} and number of ralfs is #{ralfs}"
+          if (ralfs == 0) && has_ralf
+            node.set[:clearwater][:ralf] = false
```

Comment:
`default` is a bad choice for mutable attributes, per https://github.com/pikesley/learn-yourself-some-chef/wiki/Understanding-attributes:

> So it seems that what started out as a 'default' attribute from the cookbook has been nailed in as a 'normal' value on the node, and as a 'normal' attribute has higher precedence than 'default', further changes to the 'default' cookbook attribute will be silently ignored. I presume there are good reasons for this behaviour, but whatever, my advice for now is stay away from default attributes.

(I tried setting it to "default" and saw exactly these problems.)